platform/mds/mds-web/src/main/java/org/motechproject/mds/web/service/impl/InstanceServiceImpl.java

```diff
@@ -131,6 +132,13 @@ public Object saveInstance(EntityRecord entityRecord, Long deleteValueFieldId) {
         validateNonEditableProperty(entity);
 
         List<FieldDto> entityFields = getEntityFields(entityRecord.getEntitySchemaId());
+        List<FieldRecord> fieldRecords = entityRecord.getFields();
```

Comment:
you should not set this field here, there is a setProperty method. Additionally we should support setting UUID field by end user.